Islandora/islandora_defaults
# Issue 1369: Drop Comments Dependency

**GitHub Issue**: https://github.com/Islandora/documentation/issues/1369

# What does this Pull Request do?

The comment module was hanging on as a dependency despite not being necessary. This PR adjusts the configs so we can finally be free of the dependency.

# What's new?

* Replaces the search_api and search_api_solr_defaults with search_api_solr.
* Moves the solr server and index configs from `/config/optional/` to `/config/install/`  
* Removed comment module dependency

# How should this be tested?

* View the Uninstall page any existing Islandora box and the Comment module will list several islandora-related dependencies.
* Clone the islandora-playbook
* Change [`inventory/vagrant/group_vars/webserver/drupal.yml` line 22](https://github.com/Islandora-Devops/islandora-playbook/blob/dev/inventory/vagrant/group_vars/webserver/drupal.yml#L22) from `  - "islandora/islandora_defaults:dev-8.x-1.x"` (or whatever islandora_defaults branch it targets) to `  - "islandora/islandora_defaults:dev-issue-1369"`.
* *Change the base box* E.g. `set ISLANDORA_DISTRO='ubuntu/bionic64'` OR `set ISLANDORA_DISTRO='centos/7'`.
* `vagrant up --provision` and wait.....
* Once it is up check the Uninstall page for the Comment module. *It will still not allow you to uninstall it (there is a dependency on the Article content type that Drupal installed) but at least all the islandora dependencies are gone now.*

# Additional Notes:
As mentioned in the testing instructions, this will not rid us of the Comment module; but it does make it easier. Just delete the comments field from the Article content type and you are good to go.

# Interested parties
@mjordan
